lupa/_lupa.pyx

```diff
@@ -1799,6 +1823,15 @@ cdef int py_iter_next_with_gil(lua_State* L, py_object* py_iter) with gil:
         try: runtime.store_raised_exception(L, b'error while calling next(iterator)')
         finally: return -1
 
+# keyword argument support for Python objects in Lua
+
+cdef int py_kwargs(lua_State* L) nogil:
```

Comment:
Why not construct the Python dict right here? We should do ahead of time whatever we can. If anything, then the call is the critical part, not the keyword building. Only if more than one kwargs dict is found in a call, then we need to do something (i.e. create a merged superset).

The object that we create here can be entirely opaque. Only Lupa needs to understand its intention (i.e. passing it as kwargs when found in a call).

Reply:
I disagree with having to merge kwargs dict. That just doesn't seem natural. This would also create another problem: If more than one kwargs dict is found in the call, which dict has more priority? Adding ambiguity is undesired.

```lua
python.print(1, 2, 3, python.kwargs{sep=', '}, python.kwargs{end='!\n'}, python.kwargs{sep='\t'})
```

Reply:
> I disagree with having to merge kwargs dict

The use case becomes clearer as soon as you diverge from using literal keyword arguments. Imagine a (contrived) call like
```lua
python.builtins.print(x, TO_FILE, AS_TSV)
```
where `TO_FILE = kwargs{file=some_file)` and `AS_TSV = kwargs{sep=";"}`.

> which dict has more priority

```python
>>> print(1,2,3, **dict(sep='--'), **dict(sep=':'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: print() got multiple values for keyword argument 'sep'
```

Reply:
I do understand your point now, but I need to research on how to replicate the keyword arguments expansion of multiple dictionaries or the merging of many mutually-exclusive dictionaries using the Python-C API.

Reply:
Luckily, Lupa is written in Cython, not C. :)

Reply:
You're right. We could do something in the lines of...

```cython
new_dict = dict(**dict_a, **dict_b)
```

...since `dict` is a callable so it would raise a `TypeError`.